index.js

```diff
@@ -13,7 +19,26 @@ function reducePropsToState(propsList) {
 }
 
 function handleStateChangeOnClient(stringClassNames) {
-  document.body.className = stringClassNames || '';
+  // get current body class as array
+  let currentClassNames = splitClassName(document.body.className)
+  
+  // remove all past class names
+  for (let i = 0; i < classNameCache.length; i++) {
+    const idx = currentClassNames.indexOf(classNameCache[i]);
+    if (idx !== -1) {
+      currentClassNames.splice(idx, 1);
+    }
+  }
+  
+  // append all new ones
+  const newClassNames = splitClassName(stringClassNames);
+  currentClassNames = currentClassNames.concat(newClassNames);
```

Comment:
I'm not a huge fan of the imperative-mutation approach. A functional approach would be better imho.
Maybe something like:

```js
// Lets say we already added some
BodyClassName.addedClassNames = ['foo', 'bar', 'baz'];

var currentClassNames = splitClassName(document.body.className).filter(function(
  className
) {
  return cache.indexOf(className) !== -1;
});

var newClassNames = currentClassNames
  .concat(splitClassName(stringClassNames))
  .join(' ')
  .trim();
  
BodyClassName.addedClassNames = newClassNames;
document.body.className = newClassNames;
```